crates/rome_js_analyze/src/analyzers/style/use_exponentiation_operator.rs

```diff
@@ -120,52 +82,43 @@ impl Rule for UseExponentiationOperator {
     }
 
     fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
-        let diagnostic = RuleDiagnostic::new(
+        Some(RuleDiagnostic::new(
             rule_category!(),
             ctx.query().range(),
             "Use the '**' operator instead of 'Math.pow'.",
-        );
-
-        Some(diagnostic)
+        ))
     }
 
     fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
         let node = ctx.query();
-
-        if !should_suggest_fix(node)? {
+        let mut args = node.arguments().ok()?.args().iter();
+        let (
+            Some(Ok(AnyJsCallArgument::AnyJsExpression(mut base))),
+            Some(Ok(AnyJsCallArgument::AnyJsExpression(mut exponent))),
+            None, // require no extra arguments
```

Comment:
I don't understand the third check, and the comment doesn't help. Why can't we use `node.get_arguments_by_index([0, 1]);`?

Reply:
In fact I don't get the interest of `get_arguments_by_index`. This seems overly complicated for just retrieving the two first arguments...

The third vale checks that there are only two arguments and not a third one.

Reply:
Why do you think it's overly complicated?

Reply:
> Why do you think it's overly complicated

I don't like the signature of the function:
- it is a bit weird of passing an array of indices
- It is error-prone:
  - it is easy to write `1` instead of `0` for the first argument
  - the array has to be sorted (the check is at runtime in debug mode)
- I wonder if the return type should not be `Option<[T ,N]>` instead of `[Option<T>; N]`

By way, just trying to use the function I hit a panic. The function panics when the last given index does not correspond to the last argument. I can fix it.